odc/stac/_load.py

```diff
@@ -105,7 +118,8 @@ def load(
 
      :param groupby:
         Controls what items get placed in to the same pixel plane,
-        supported values are "time" or "solar_day", default is "time"
+        supported values are "time", "solar_day" and "nothing",
```

Comment:
I agree that "nothing" is not a great name for this. We should find some other way to indicate that. Using `None` for this would make sense if it was the default behaviour. But default behaviour is to group by timestamp, and I think it's what most users would expect/want, but not grouping at all is also useful sometimes. It just feels odd to demand `(..., groupby=None)` to achieve that, as `=None` is so strongly associated with "default" in Python.

One option is to use `groupby="id"`, which is kinda true since "id" is supposed to be unique, issue with that is "edge cases", what if you do have duplicate ids?

Maybe `groupby=":ungrouped:"`, with `:` indicating that it's "special", or even just `groupby=""`?

Reply:
> as =None is so strongly associated with "default" in Python.

I don't know if I agree -- while it _can_ be default, `Optional[Thing]` is absolutely a pattern as well.

If we are going to use a special string to indicate absence, then I'd probably lobby for an `Enum` to make the different cases more explicit.

Reply:
`Optional[Thing]` is a thing for sure, but `some_arg: Optional[Thing] = DefaultThingThatIsNotNone` is less of a thing.

Enum would be ok if this parameter was not called `groupby=`, this is somewhat modeled by pandas groupby and parameter there is `mapping, function, label, or list of labels`, this currently only supports `label` version of that, where `time, solar_day` are a kind of virtual properties of items. I'm starting to like `"id"` more as it fits this semantic. 

Obviously pandas groupby does not need to support "single element groups only mode" as you simply don't call it if you don't need it. I guess `groupby=None` could mean "don't do grouping at all", only things is accepting `=None` without defaulting to it feels odd.

Reply:
> only things is accepting =None without defaulting to it feels odd.

Yeah I guess we'll differ here but that's alright :-). `"id"` does feel better than `"nothing"`.